cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -309,8 +296,14 @@ def start_kube_dns():
             'dns_domain': hookenv.config('dns_domain')
         }
     }
-    create_addon('kubedns-controller.yaml', context)
-    create_addon('kubedns-svc.yaml', context)
+
+    try:
+        create_addon('kubedns-controller.yaml', context)
+        create_addon('kubedns-svc.yaml', context)
+    except CalledProcessError:
+        hookenv.log('kubedns deploy failed, will try again soon')
```

Comment:
Good idea, though it's hard to give a good message that explains what we're waiting for. Usually `CalledProcessError` just means the API server isn't fully up yet, but it can mean other things too.

Probably just go with "Waiting to retry KubeDNS deployment" for now. I'll get that in there.